### PR TITLE
Template set deployables tests

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Adresses/Fixes #1452.

@devonfw/cobigen

### CODE REVIEW

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java [hunk 1]
```diff
@@ -73,35 +73,36 @@ public void cleanup() {
...
 
-    Path cobigenTemplatesParent = this.cobiGenHome.toPath()
-        .resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATES_PATH);
+    Path cobigenTemplatesParent = this.cobiGenHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATES_PATH);
     Files.createDirectories(cobigenTemplatesParent);
     Path cobigenTemplatesProject = cobigenTemplatesParent.resolve(ConfigurationConstants.COBIGEN_TEMPLATES);
     Files.createDirectories(cobigenTemplatesProject);
     CobiGenFactory.extractTemplates();
-    assertThat(cobigenTemplatesProject.toFile()).exists().isDirectory();
+    assertThat(Files.isDirectory(cobigenTemplatesProject));
```
Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: again it's better to put isDirectory() at the end

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java [hunk 2]
```diff
@@ -73,35 +73,36 @@ public void cleanup() {
...
     CobiGenFactory.extractTemplates();
-    Path adaptedFolder = this.cobiGenHome.toPath().resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH)
+    Path adaptedFolder = this.cobiGenHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH)
         .resolve(ConfigurationConstants.ADAPTED_FOLDER);
     Path extractedJar1 = adaptedFolder.resolve("template-test1-0.0.1");
     Path extractedJar2 = adaptedFolder.resolve("template-test2-0.0.1");
-    assertThat(extractedJar1.toFile()).exists().isDirectory();
-    assertThat(extractedJar2.toFile()).exists().isDirectory();
+    assertThat(Files.isDirectory(extractedJar1));
+    assertThat(Files.isDirectory(extractedJar2));
```
Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: I think it's better to use:
assertThat(extractedJar.exists()).isDirectory();
that way you get a more accurate error message.

**Review by LarsReinken — CHANGES_REQUESTED** [review]

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java [hunk 3]
```diff
@@ -73,35 +73,36 @@ public void cleanup() {
...
   }
 
   /**
+   * Test of extract templates with old CobiGen_Templates project existing
+   *
    * @throws IOException if an Exception occurs
    */
-  @Test
   public void extractTemplatesWithOldConfiguration() throws IOException {
```
Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java:98]: this method is just used in this class and should be private.

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java [hunk 4]
```diff
@@ -73,35 +73,36 @@ public void cleanup() {
   }
 
   /**
+   * Tests if template sets can be extracted properly
+   *
    * @throws IOException if an Exception occurs
    */
-  @Test
   public void extractTemplateSetsTest() throws IOException {
```
Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java:80]: this method is just used in this class and should be private.

Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: this method should be private if it isn't a test.

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java (hunk 3 above)
Comment (LarsReinken) [inline cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: this method should be private if it isn't a test.

**Review by LarsReinken — CHANGES_REQUESTED** [review]

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java (hunk 4 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: Adjusted.

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java (hunk 3 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: Adjusted.

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java (hunk 2 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: Adjusted.

File: cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java (hunk 1 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java]: Adjusted.

**Pull request merged**